distribution/lib/Standard/Table/0.0.0-dev/src/Data/Column.enso

```diff
@@ -732,32 +732,32 @@ type Column
             if self.value_type.is_integer && decimal_places >= 0 then self.rename new_name else
                 should_cast_to_integer = decimal_places <= 0 || self.value_type.is_integer
                 target_value_type = if should_cast_to_integer then Value_Type.Integer else self.value_type
-                result = self.check_round_input target_value_type <|
+                result = self.check_round_input col->
                     scale = 10 ^ decimal_places
-                    scaled = self * scale
+                    scaled = col * scale
                     round_base = scaled.floor
                     round_midpoint = (round_base + 0.5) / scale
-                    even_is_up = (self >= 0).iif ((scaled.truncate % 2) != 0) ((scaled.truncate % 2) == 0)
-                    half_goes_up = if use_bankers then even_is_up else self >= 0
-                    do_round_up = half_goes_up.iif (self >= round_midpoint) (self > round_midpoint)
+                    even_is_up = (col >= 0).iif ((scaled.truncate % 2) != 0) ((scaled.truncate % 2) == 0)
+                    half_goes_up = if use_bankers then even_is_up else col >= 0
+                    do_round_up = half_goes_up.iif (col >= round_midpoint) (col > round_midpoint)
                     result_float = do_round_up.iif ((round_base + 1.0) / scale) (round_base / scale)
                     cast_if_needed result_float target_value_type
                 result.rename new_name
 
     ## PRIVATE
        Restrict allowed range of input to rounding methods.
-    check_round_input : Column -> Value_Type -> Any ! Illegal_Argument
-    check_round_input self target_value_type column =
+    check_round_input : (Column -> Column) -> Any ! Illegal_Argument
+    check_round_input self action =
         min = Rounding_Helpers.round_min_long
         max = Rounding_Helpers.round_max_long
-        within_range = self >= self.const min && self <= self.const max
-        warning = Illegal_Argument.Error <| "Error: `round` can only accept values between " + min.to_text + " and " + max.to_text + " (inclusive)"
-        nothing_with_problem = Warning.attach warning Nothing
-        ## TODO: This should be cast to NULL when that type is available,
-           see https://github.com/enso-org/enso/issues/6281.
-        nothings = self.const nothing_with_problem . cast target_value_type
-        result = within_range.iif column nothings
-        result.rename column.name
+        within_range = if self.value_type.is_floating_point.not then self.between min max else
+            self.is_nan || self.is_infinite || self.between min max
+
+        any_issues = within_range.to_vector . any (== False)
+        if any_issues.not then action self else
+            warning = Illegal_Argument.Error <| "Error: `round` can only accept values between " + min.to_text + " and " + max.to_text + " (inclusive)"
+            result = action (within_range.iif self Nothing)
```

Comment:
Since we're already converting to vector and traversing the column, we could just build up the result with results and Nothings at the same time, to save an extra pass.

Reply:
Sounds good but is that possible without changing more code? I tihnk then we cannot easily use the vectorized operations